odoo/addons/base/models/res_company.py

```diff
@@ -340,6 +340,15 @@ def unlink(self):
         return res
 
     def write(self, values):
+        if 'parent_id' in values:
+            raise UserError(_("The company hierarchy cannot be changed."))
+
+        if values.get('currency_id'):
+            currency = self.env['res.currency'].browse(values['currency_id'])
+            if not currency.active:
+                currency.write({'active': True})
+
+        res = super().write(values)
```

Comment:
Could you add a test? And I am not sure if the cache invalidation in `create` will have cause same problem